test/unit/graphql_test.lua

```diff
@@ -1092,3 +1092,18 @@ end
 g.test_version = function()
     t.assert_equals(require('graphql').VERSION, 'scm-1')
 end
+
+function g.test_is_array()
+    t.assert_equals(util.is_array({[3] = 'a', [1] = 'b', [6] = 'c'}), true)
+    t.assert_equals(util.is_array({[3] = 'a', [1] = 'b', [7] = 'c'}), true)
+    t.assert_equals(util.is_array({[3] = 'a', nil, [6] = 'c'}), true)
+    t.assert_equals(util.is_array({[3] = 'a', nil, [7] = 'c'}), true)
+    t.assert_equals(util.is_array({[0] = 'a', [1] = 'b', [6] = 'c'}), true)
+    t.assert_equals(util.is_array({[-1] = 'a', [1] = 'b', [6] = 'c'}), true)
```

Comment:
It's a very controversial topic. I don't see any reasons to change anything. I don't see array definition or any reasonable description in commit message.

I don't understand also why table that contains negative elements is array.
Also probably we could consider __serialize property of metatable.

Reply:
Facts:
- the problem is described in the linked issue, can't see any reasons to copy-paste.
- is_array() works not properly and I suggested one of possible solutions.

Reasons to fix:
graphql.util.is_array() used not only in this modue but also in apps code since it available in public API, and I see only 3 upshots:
- fix it anyway
- add clear description of is_array() behaviour to save developer's time
- remove it from public API of graphql lib